cockpit/src/subscriptions-client.js

```diff
@@ -124,161 +92,104 @@ function getSubscriptionDetails() {
-    if (subscriptionDetails.user || subscriptionDetails.password) {
-        if (!subscriptionDetails.user)
-            subscriptionDetails.user = '';
-        if (!subscriptionDetails.password)
-            subscriptionDetails.password = '';
-        args.push('--username', subscriptionDetails.user, '--password', subscriptionDetails.password);
+    if (subscriptionDetails.url != 'default') {
+        /*  parse url into host, port, handler; sorry about the ugly regex
+            (?:https?://)? strips off the protocol if it exists
+            ([$/:]+) matches the hostname
+            (?::(?=[0-9])([0-9]+))? matches the port if it exists
+            (?:(/.+)) matches the rest for the path
+        */
+        pattern = new RegExp('^(?:https?://)?([^/:]+)(?::(?=[0-9])([0-9]+))?(?:(/.+))?$');
+        match = pattern.exec(subscriptionDetails.url); // TODO handle failure
+        options.host = {
+            t: 's',
+            v: match[1],
+        };
+        options.port = {
+            t: 's',
+            v: match[2],
+        };
+        options.handler = {
+            t: 's',
+            v: match[3],
+        };
     }
 
+    var connection_options = {};
     // proxy is optional
     if (subscriptionDetails.proxy) {
-        if (!subscriptionDetails.proxyServer)
-            subscriptionDetails.proxyServer = '';
-        if (!subscriptionDetails.proxyUser)
-            subscriptionDetails.proxyUser = '';
-        if (!subscriptionDetails.proxyPass)
-            subscriptionDetails.proxyPass = '';
-        args.push('--proxy', subscriptionDetails.proxyServer,
-                  '--proxyuser', subscriptionDetails.proxyUser,
-                  '--proxypass', subscriptionDetails.proxyPass);
+        connection_options.proxy_hostname = {
+            t: 's',
+            v: subscriptionDetails.proxyServer,
+        };
+        connection_options.proxy_user = {
+            t: 's',
+            v: subscriptionDetails.proxyUser,
+        }
+        connection_options.proxy_password = {
+            t: 's',
+            v: subscriptionDetails.proxyPass,
+        }
     }
 
-    // only pass org info if user provided it
-    if (subscriptionDetails.org)
-        args.push('--org', subscriptionDetails.org);
-
-    /* TODO DBus API doesn't deliver what we need, so we call subscription manager
-     * without translations and parse the output
-     * https://bugzilla.redhat.com/show_bug.cgi?id=1304056
-     */
-    var process = cockpit.spawn(args, {
-        directory: '/',
-        superuser: "require",
-        environ: ['LC_ALL=C'],
-        err: "out"
-    });
-
-    var promise;
-    var buffer = '';
-    process
-        .input('')
-        .stream(function(text) {
-            buffer += text;
-        })
-        .done(function(output) {
-            dfd.resolve();
-        })
-        .fail(function(ex) {
-            if (ex.problem === "cancelled") {
-                dfd.reject(ex);
-                return;
+    registerServer.Start()
```

Comment:
You may wind up with a bug here.  When you create a proxy, since it is asynchronous, you don't know when the proxy object will actually be ready.  I have seen this before, where just calling Start() right away will work when I test it in the javascript console but not as a script (because there is a latency for me to type in commands in the console).  

There's a proxy.valid property you can check, or you can call proxy.wait() where wait() can take a function which will be called when the proxy is ready, and wait() will return a Promise.  So you probably want to do this:

```javascript
let regSrvProm = registerServer.wait();
regSrvProm
  .then(() => {
      registerServer.Start()
          .then(function(socket) {
              // ...
          })
  })
```

or 

```javascript
let regSrvProm = registerServer.wait(() => {
    registerServer.Start()
        .then(socket => {
            // ...
        })
})
```